src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp

```diff
@@ -348,7 +350,8 @@ struct layout {
     }
 
     friend bool operator==(const layout& lhs, const layout& rhs) {
-        return lhs.data_type == rhs.data_type && lhs.format == rhs.format && lhs.size == rhs.size && lhs.data_padding == rhs.data_padding;
+        auto check_layout = (lhs.is_dynamic() || rhs.is_dynamic()) ? (lhs.size == rhs.size) : (lhs.get_dims() == rhs.get_dims());
```

Comment:
little bit confused becuase in the original static scenario (master), it was checking size. However here it checks size when the layout is dynamic. Could you explain?

Reply:
In this commit, there is case where the format of input and output of layer(e.g. reshape) is same, but the rank of dimension is different and fusing couldn't possible. So for now, I reverted this commit (https://github.com/andrew-k-park/openvino/commit/8616eaccac3da9e1cb1ffa56b411906b2079b794) and will fix partial shape comparison related issues with a separate PR.